File: Jasmine/JasmineCompiler/Src/DFIR_Compiler/DFIR_Compiler/IDataTypeBuilder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DFIR_Compiler
{
    public class IDataTypeBuilder
    {
        private IDataType _IDataType;
        public IDataTypeBuilder(BaseDataType _base)
        {
            _IDataType = new IDataType(_base);
        }
        public virtual void SetWordLength(int WordLength)
        {
            _IDataType.setWordLength(WordLength);
        }
        public virtual void SetLeftLength(int LeftLength)
        {
            _IDataType.setLeftLength(LeftLength);
        }
        public virtual void SetRightLength(int RightLength)
        {
            _IDataType.setRightLength(RightLength);
        }
        public virtual void SetDimensions(int Dimensions)
        {
            _IDataType.setDimensions(Dimensions);
        }
        public virtual void SetArraySize(int ArraySize)
        {
            _IDataType.setArraySize(ArraySize);
        }
        public virtual void SetElementType(IDataType ElementType)
        {
            _IDataType.setElementType(ElementType);
        }
        public virtual IDataType GetResult()
        {
            return _IDataType;
        }

    }
}

File: Wayne/WayneCompiler/src/Dfir_representation/IDataTypeBuilder.java
package Dfir_representation;

public  class IDataTypeBuilder{
    
    private IDataType _IDataType;
 	public IDataTypeBuilder(BaseDataType base){
	 	     this._IDataType = new IDataType(base);
	    }
	public void SetWordLength(int WordLength){
		     this._IDataType.setWordLength(WordLength);
	    }
  	public void SetLeftLength(int LeftLength){
		     this._IDataType.setLeftLength(LeftLength);
	    }
	public void SetRightLength(int RightLength){
		     this._IDataType.setRightLength(RightLength);
	    }
	public void SetDimensions(int Dimensions){
		     this._IDataType.setDimensions(Dimensions);
	    }
	public void SetArraySize(int ArraySize){
		     this._IDataType.setArraySize(ArraySize);
	    }
	public void SetElementType(IDataType ElementType){
		     this._IDataType.setElementType(ElementType);
	    }
	public IDataType GetResult(){
	    	return this._IDataType;
	    }
}